### ISSUE TITLE
scala-maven-plugin: "character to be escaped is missing" on win7/x64

### ISSUE THREAD

**Issue by danieldietrich** [javaslang/javaslang#277]
@chagmed reported an UTF-8 issue with building Javaslang 1.2.3 on a Win7-x64 box. The scala-maven-plugin 3.2.0 reported: "character to be escaped is missing". More specifically: "IO error while decoding javaslang\target\.scalaScriptGen\Generator_1.scala with UTF-8"

**Comment by danieldietrich** [comment]
Are there any win7/x64 users here which are able to compile the Javaslang Github repo on their machine via maven? Need some feedback!

**Comment by danieldietrich** [comment]
I could reproduce this using VirtualBox with Win7/x64, Java 1.8.0_45, git-bash and maven 3.3.3.

Setting the scalac file encoding in the pom.xml did not help:

```shell
<args>
    <arg>-encoding</arg>
    <arg>UTF8</arg>arg>
</args>
```

This needs to be further investigated.

**Comment by ignasi35** [comment]
Hi @danieldietrich ,  

I did a quick native2ascii and apart from comments there's a ' lambda-sign' that got escaped. Could that be it?

```
(excerpt)
712c712
<           public interface $className$fullGenerics extends λ<R>$additionalExtends {
---
>           public interface $className$fullGenerics extends \u03bb<R>$additionalExtends {
```

Sorry I can't test it on Win machine at the moment.

Cheers,




PS: full diff:

```

511[generator](master)$ /Library/Java/JavaVirtualMachines/jdk1.8.0_40.jdk/Contents/Home/bin/native2ascii Generator.scala > out

512[generator](master)$ diff Generator.scala out
642c642
<               // ¬(p => q) ≡ ¬(¬p ∨ q) ≡ p ∧ ¬q
---
>               // \u00ac(p => q) \u2261 \u00ac(\u00acp \u2228 q) \u2261 p \u2227 \u00acq
712c712
<           public interface $className$fullGenerics extends λ<R>$additionalExtends {
---
>           public interface $className$fullGenerics extends \u03bb<R>$additionalExtends {
1197c1197
<                 // (∀a,b ∈ ℝ+ ∃c ∈ ℝ+ : a²+b²=c²) ≡ (∀a,b ∈ ℝ+ : √(a²+b²) ∈ ℝ+)
---
>                 // (\u2200a,b \u2208 \u211d+ \u2203c \u2208 \u211d+ : a\u00b2+b\u00b2=c\u00b2) \u2261 (\u2200a,b \u2208 \u211d+ : \u221a(a\u00b2+b\u00b2) \u2208 \u211d+)
1207c1207
<                 // ∀is,ss: length(is) = length(ss) → unzip(zip(is, ss)) = (is, ss)
---
>                 // \u2200is,ss: length(is) = length(ss) \u2192 unzip(zip(is, ss)) = (is, ss)

```

**Comment by danieldietrich** [comment]
Hi @ignasi35,

thanks for the tests!

Yes, these are the candidates. I think the so called 'script mode' of the scala-maven-plugin accidently messes up the encoding when creating a copy of Generator.scala in the /target dir of the project.

I will contact a scala-maven-plugin dev...

Greets!

**Comment by danieldietrich** [comment]
I filed a bug davidB/scala-maven-plugin#168

**Comment by danieldietrich** [comment]
Should be fixed with scala-maven-plugin 3.2.1 but does not seem to. Tested it under Win, the error still remains. Have to further investigate it...
